Set extra result data fields

bram123 opened the pull request:
|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | no
| BC Break      | no
| New Feature   | no
| RFC           | no
| QA            | yes

### Description

Adding extra Check Result `data` values for some checks where this was missing.
Created new ExtensionsLoadedTest, assumes that xml/json extensions exists (these are required by phpunit itself).

bram123 committed:
Set extra result data fields

Signed-off-by: Bram Leeda <bram@123inkt.nl>

Ocramius commented on test/ExtensionLoadedTest.php:
```diff
@@ -0,0 +1,63 @@
+        self::assertSame(['json' => phpversion('json'), 'xml' => phpversion('xml')], $result->getData());
+    }
+
+    public function invalidArgumentProvider(): array
```
This method requires type documentation

bram123 replied:
Added typehints in the docblock, picked `mixed` to not be stuck to int|stdClass for later additions to the dataprovider.
Had to update the psalm baseline, because psalm detects that the test is sending invalid types to the constructor, but this is the point of the unittest.

Ocramius commented on test/ExtensionLoadedTest.php:
```diff
@@ -0,0 +1,63 @@
+        new ExtensionLoaded($extensionName);
+    }
+
+    public function testCheck(): void
+    {
+        $check  = new ExtensionLoaded('json');
+        $result = $check->check();
+
+        self::assertInstanceof(SuccessInterface::class, $result);
+        self::assertSame('json ' . phpversion('json'), $result->getData());
+
+        $check  = new ExtensionLoaded('unknown-foo');
+        $result = $check->check();
+
+        self::assertInstanceof(FailureInterface::class, $result);
+        self::assertSame(['unknown-foo'], $result->getData());
+
+        $check  = new ExtensionLoaded(['unknown-foo', 'unknown-bar']);
+        $result = $check->check();
+
+        self::assertInstanceof(FailureInterface::class, $result);
+        self::assertSame(['unknown-foo', 'unknown-bar'], $result->getData());
+
+        $check  = new ExtensionLoaded(['json', 'xml']);
+        $result = $check->check();
+
+        self::assertInstanceof(SuccessInterface::class, $result);
+        self::assertSame(['json' => phpversion('json'), 'xml' => phpversion('xml')], $result->getData());
+    }
```
Let's please have different test methods for these cases

Ocramius reviewed (changes requested):
The approach seems valid: adjustments needed, but can then be merged + shipped :+1:

bram123 committed:
Split ExtensionLoadedTest testCheck to have a separate testCheckMissingExtension method.
Add return docblock to invalidArgumentProvider.

Signed-off-by: Bram Leeda <bram@123inkt.nl>

bram123 committed:
Remove unused import from ExtensionLoadedTest.php
Add invalid type issue to psalm-baseline.xml
- the unittest is sending invalid types on purpose to test the constructors validation

Signed-off-by: Bram Leeda <bram@123inkt.nl>

Ocramius reviewed (approved):
LGTM :+1:

Thanks @bram123!

bram123 commented:
Thanks for the very quick review @Ocramius!

Ocramius commented:
@bram123 is also released :+1:
